src/functions.php

```diff
@@ -377,13 +381,104 @@ function upload(string $xml, $config)
         'PhonebookImportFile' => [
             'type' => 'text/xml',
             'filename' => 'updatepb.xml',
-            'content' => $xml,
+            'content' => $XMLNewPhonebook->asXML(), // convert XML object to XML string
         ]
     ];
 
-    $result = $fritz->postFile($formfields, $filefields); // send the command
-
+    $result = $fritz->postFile($formfields, $filefields); // send the command to store new phonebook
     if (strpos($result, 'Das Telefonbuch der FRITZ!Box wurde wiederhergestellt') === false) {
         throw new \Exception('Upload failed');
     }
 }
+
+
+/**
+ * Downloads the old phone book from Fritzbox
+ *
+ * @param   Api   $fritz
+ * @param   array $config
+ * @return  SimpleXMLElement with the old phonebook
+ */
+function getOldPhonebook(Api $fritz, array $config)
+{
+    $formfields = [
+        'PhonebookId' => $config['phonebook']['id'],
+        'PhonebookExportName' => $config['phonebook']['name'],
+        'PhonebookExport' => "",
+    ];
+    $result = $fritz->postFile($formfields, []); // send the command to load existing phone book
+    if (substr($result, 0, 5) !== "<?xml") {
```

Comment:
XML checking should be done by parsing and capturing an exception.

Reply:
I copied this line  from another FritzBox project (of Mr. Maus I think). 
But could you give me one more hint?
I just want to find out of ANY XML comes back.

Reply:
this will work fine unless somebody throws broken xml at it (or maybe even doctype declarations?). If you want to change it there's `simplexml_load_string`

Reply:
Hmmm. I gave it a try... and a few tests...
Problem here is: FritzBox is inconistent: in case of succes it deliveres `<?xml ...` string in case of problems (e.g. no login), it delivers HTML like so: `<!DOCTYPE html><html lang="de">...` - which renders a normal FB login screen.
If I feed this HTML result string to `simplexml_load_string` I see loads of PHP warning on the console, **but no exception!**

```
PHP Warning:  simplexml_load_string(): Entity: line 4: parser error : AttValue: " or ' expected in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
PHP Warning:  simplexml_load_string(): <meta http-equiv=content-type content="text/html; charset=utf-8" /> in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
PHP Warning:  simplexml_load_string():                  ^ in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
PHP Warning:  simplexml_load_string(): Entity: line 4: parser error : attributes construct error in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
PHP Warning:  simplexml_load_string(): <meta http-equiv=content-type content="text/html; charset=utf-8" /> in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
PHP Warning:  simplexml_load_string():                  ^ in /home/pi/bin/carddav2fb_dev/src/functions.php on line 410
``` 

So, I would rather stick to the code above which - well - just works.
If FB changes behaviour here, we will see it quickly, I think.